### DISCUSSION TITLE
Mobile menu theme and language configuration

### DISCUSSION THREAD

**Discussion by magicDGS** [withastro/starlight#979] [Feature Requests]
### What version of `starlight` are you using?

0.11.1

### What is your idea?

Currently on mobile the menu shows the theme and  language on the footer (and maybe the social icons, if they are shown there after https://github.com/withastro/starlight/issues/978). The idea would be to configure if those components are in a header or a footer (or any other solution to fix the issue described below).

### Why is this feature necessary?

For small size documentation, the current behavior might be fine, but for large documentation the user might need to scroll to the end just to see those options.

In addition, this combined with the problems described in https://github.com/withastro/starlight/issues/977 and https://github.com/withastro/starlight/issues/978 are making the experience with mobile a bit clunky.

Last but not least, if an extension like starlight-blog (see also https://github.com/withastro/starlight/discussions/948) is used and the component still behaves in the same way (footer), accessing the blog from mobile (more common than accessing the full documentation) might be difficult.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by newtoallofthis123** [comment]
Interesting idea! I had a similar problem as well 😅

**Comment by HiDeoo** [comment]
Not to say the idea is not interesting, on the contrary, but I wanted to mention that [component overrides](https://starlight.astro.build/guides/overriding-components/) already allow for users to potentially override this behavior if this is something blocking/annoying for them to wait for a potential built-in solution to address this.

For example, in `starlight-blog` as it was mentioned, we use that to always render the blog link above the sidebar content on mobile:

<img width="328" alt="image" src="https://github.com/withastro/starlight/assets/494699/a28b496f-08c0-43da-9653-149d9a4e544e">

**Reply by magicDGS** [reply]
Thanks for the hint. I will use that in the meantime!

**Comment by delucis** [comment]
Thanks for the suggestion! I wonder if making that footer section sticky could be a good solution instead of a configuration option? That way, the language/theme options would always be visible and just the navigation links scroll.

As @HiDeoo says, this should be possible today with component overrides, but definitely something to think about whether we can also improve the default behaviour.

**Comment by magicDGS** [comment]
Thanks for the response. Making it sticky sounds like a better solution, I was just suggesting a configuration not to be intrusive in the current behavior. I am kind of new on terms of UI/UX and accessibility, so I don't feel that I have the knowledge to do strong suggestions such that.

**Comment by Devarajan89** [comment]
![Screenshot 2025-07-04 at 11 47 36](https://github.com/user-attachments/assets/b76939a9-2532-423f-be11-5a907945ecf1)
Overided the component to show it on top
